src/counts.jl

```diff
@@ -231,22 +231,30 @@ end
 
 
 """
-    addcounts!(dict, x[, wv])
+    addcounts!(dict, x[, wv]; alg = :auto)
 
 Add counts based on `x` to a count map. New entries will be added if new values come up.
 If a weighting vector `wv` is specified, the sum of the weights is used rather than the
 raw counts.
+
+alg can be one of
+    :auto       -   default; uses :radixsort if safe to do so
+    :radixsort  -   if radixsort_safe(eltype(x)) == true then use radixsort to
+                    sort the input array in order to achieve faster performance
+                    by using more RAM. One should choose :dict if the
+                    amount of available RAM is a limitation.
+    :dict       -   Uses Dict-based method which is generally slower but uses
+                    less RAM and is safe for any data type.
 """
-function addcounts!(cm::Dict{T}, x::AbstractArray{T}) where T
+function addcounts!(cm::Dict{T}, x::AbstractArray{T}; alg = :auto) where T
     # if it's safe to be sorted using radixsort then it should be faster
     # albeit using more RAM
-    if radixsort_safe(T)
+    if radixsort_safe(T) & (alg == :auto | alg == :radixsort)
```

Comment:
Use `&&` and `||`, the compiler is smart enough to decide whether to use a branch or not.